lib/offsite_payments/integrations/paytm.rb

```diff
@@ -0,0 +1,298 @@
+module OffsitePayments #:nodoc:
+  module Integrations #:nodoc:
+    module Paytm
+      require 'openssl'
+      require 'base64'
+      require 'digest'
+      require 'securerandom'
+
+      mattr_accessor :test_url
+      mattr_accessor :production_url
+
+      self.test_url = 'https://pguat.paytm.com/oltp-web/processTransaction'
+      self.production_url = 'https://secure.paytm.in/oltp-web/processTransaction'
+
+      def self.service_url
+        # self.production_url
+        OffsitePayments.mode == :production ? production_url : test_url
+      end
+
+      def self.notification(post, options = {})
+        Notification.new(post, options)
+      end
+
+      def self.return(post, options = {})
+        Return.new(post, options)
+      end
+
+      def self.checksum(params, secret_key, salt = nil)
+        salt = SecureRandom.urlsafe_base64(4 * (3 / 4)) if salt.nil?
+        keys = params.keys
```

Comment:
Hi Gaurav,

Can you please update the status.

Regards,

On Thu, Aug 18, 2016 at 3:04 AM, Gaurav Chande <notifications@github.com>
wrote:

> In lib/offsite_payments/integrations/paytm.rb
> <https://github.com/activemerchant/offsite_payments/pull/200#discussion_r75211861>
> :
>
> > +      def self.service_url
> > +        # self.production_url
> > +        OffsitePayments.mode == :production ? production_url : test_url
> > +      end
> > +
> > +      def self.notification(post, options = {})
> > +        Notification.new(post, options)
> > +      end
> > +
> > +      def self.return(post, options = {})
> > +        Return.new(post, options)
> > +      end
> > +
> > +      def self.checksum(params, secret_key, salt = nil)
> > +        salt = SecureRandom.urlsafe_base64(4 * (3 / 4)) if salt.nil?
> > +        keys = params.keys
>
> I tried running the gateway with sandbox credentials but it failed at this
> step. form_fields method errors out saying: undefined method 'keys' for
> #<Array:0x007fbae0c3e2c0>. Your code is doing Paytm.checksum(checksum_payload_items,
> @options[:credential2]) in a previous step and checksum_payload_items is
> actually an array, not a hash, so params.keys will fail.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/activemerchant/offsite_payments/pull/200/files/325209e8a55619ef7c5f6064c53a75802a69ad93#r75211861>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AL1PqGSgmEl_LDU7FBcgtlsR2npBjF4gks5qg35ygaJpZM4IQnF4>
> .
>



-- 
*Mohit Aggarwal*
Senior Software Engineer
+91 8826036707
[image: Inline image 1]
www.Paytm.com

-- 


Disclaimer :-

This email and any files transmitted with it are confidential and 
intended solely for the use of the individual or entity to which they are 
addressed. If you are not the intended recipient, you should not 
disseminate, distribute or copy this e-mail. Please notify the sender 
immediately by e-mail and destroy all copies of this message and any 
attachments. Any views or opinions presented in this email are solely those 
of the author and do not necessarily represent those of the company. 

Warning: Although the company has taken reasonable precautions to ensure no 
viruses are present in this email, the company cannot accept responsibility 
for any loss or damage arising from the use of this email or attachments.